src/Jaeger.Core/Reporters/RemoteReporter.cs

```diff
@@ -123,14 +123,14 @@ private async Task FlushLoop()
             while (!_commandQueue.IsAddingCompleted);
         }
 
-        private void ProcessQueueLoop()
+        private async Task ProcessQueueLoop()
         {
             // This blocks until a command is available or IsCompleted=true
             foreach (ICommand command in _commandQueue.GetConsumingEnumerable())
```

Comment:
Hi @phxnsharp ,

I agree that GetConsumingEnumerable() is blocking the code. Also I agree that this will not fix #189, and that #189 was not caused by #182, that's why I closed this merge request.

I already created a new merge request #191 which is using BufferBlock instead of blocking collection. IF you can take a look at it that would be great.